### PR TITLE
Update STAR/STARsolo to latest version and fix index data table

### PR DESCRIPTION
**Pull request by mtekman**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [X] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Based on: https://github.com/galaxyproject/tools-iuc/pull/2648

### CODE REVIEW

File: data_managers/data_manager_star_index_builder/data_manager_conf.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <data_managers>
-    <data_manager tool_file="data_manager/rna_star_index_builder.xml" id="rna_star_index_builder" version="0.0.6">
-        <data_table name="rnastar_index2_versioned">
+    <data_manager tool_file="data_manager/rna_star_index_builder.xml" id="rna_star_index_builder" version="0.0.7">
+        <data_table name="rnastar_index2x_versioned">
```
Comment (bgruening) [inline data_managers/data_manager_star_index_builder/data_manager_conf.xml:4]: Is there any need to rename the data table?

Comment (wm75) [reply data_managers/data_manager_star_index_builder/data_manager_conf.xml:4]: Yes, that's the core part of all of this. Required to fix https://github.com/galaxyproject/tools-iuc/pull/2637

Comment (wm75) [reply data_managers/data_manager_star_index_builder/data_manager_conf.xml:4]: I really hope it's the last time :)

Comment (bgruening) [reply data_managers/data_manager_star_index_builder/data_manager_conf.xml:4]: I see. Please let me know when this gets in. I guess I need to install a ton of genomes again ;)

**Comment by wm75** [comment]
@mtekman do I understand correctly that STARsolo does not work without a gene model in the index?
If that's true, should we remove the option to build/use an index without gene model then?

**Comment by mtekman** [comment]
@wm75 yes that's correct, and yes it makes sense to remove it

**Comment by wm75** [comment]
@mtekman This looks ready to me. If you still want to add an option to produce BAM output from STARsolo, go ahead.
Otherwise, we can merge this tomorrow.

File: tools/rgrnastar/rg_rnaStarSolo.xml [hunk 2]
```diff
@@ -73,6 +73,13 @@
     --soloStrand $solo.soloStrand
     --soloFeatures $solo.soloFeatures
     --soloUMIdedup $solo.soloUMIdedup
+    --quantMode TranscriptomeSAM
+    ## choosing BAM SortedByCoord appears to give the *less* reads
```
Comment (bgruening) [inline tools/rgrnastar/rg_rnaStarSolo.xml]: Wow, this is scary! There is a *the* too many in this sentence.

Comment (mtekman) [reply tools/rgrnastar/rg_rnaStarSolo.xml]: fixed :-)

File: tools/rgrnastar/rg_rnaStarSolo.xml [hunk 3]
```diff
@@ -148,204 +167,194 @@
...
-        <data format="mtx" name="output_matrixSJ" label="${tool.name} on ${on_string}: Matrix Splice Junction Counts" from_work_dir="Solo.out/matrixSJ.mtx" >
+        <data format="mtx" name="output_matrixSJ" label="${tool.name} on ${on_string}: Matrix Splice Junction Counts" from_work_dir="Solo.out/Gene/filtered/matrixSJ.mtx" >
             <filter>solo['soloFeatures'] == "SJ" </filter>
         </data>
-        <data format="mtx" name="output_matrixGeneFull" label="${tool.name} on ${on_string}: Matrix Full Gene Counts" from_work_dir="Solo.out/matrixGeneFull.mtx" >
+        <data format="mtx" name="output_matrixGeneFull" label="${tool.name} on ${on_string}: Matrix Full Gene Counts" from_work_dir="Solo.out/Gene/filtered/matrixGeneFull.mtx" >
             <filter>solo['soloFeatures'] == "GeneFull" </filter>
         </data>
-        <data format="txt" name="output_stats" label="${tool.name} on ${on_string}: Feature Statistic Summaries" from_work_dir="Solo.out/Gene.stats" />
+        <data format="bam" name="output_BAM" label="${tool.name} on ${on_string}: Alignments" from_work_dir="Aligned.sorted.bam" />
```
Comment (bgruening) [inline tools/rgrnastar/rg_rnaStarSolo.xml:181]: You add a dbKey with `dbKeyActions` to the log file above. If this is still working, maybe this is also useful for the BAM file various other mtx files as well?

File: tools/rgrnastar/rg_rnaStarSolo.xml [hunk 4]
```diff
@@ -73,6 +73,14 @@
     --soloStrand $solo.soloStrand
     --soloFeatures $solo.soloFeatures
     --soloUMIdedup $solo.soloUMIdedup
+    --quantMode TranscriptomeSAM
+    --outSAMtype BAM Unsorted
+
+    && samtools sort Aligned.out.bam -o Aligned.sorted.bam
```
Comment (bgruening) [inline tools/rgrnastar/rg_rnaStarSolo.xml:79]: you can and should use multiple cores and tmp dir ...

https://github.com/galaxyproject/tools-iuc/blob/ffcf3989e7b05726341bbd0457a2e58448361554/tools/minimap2/minimap2.xml#L162

File: tools/rgrnastar/rg_rnaStarSolo.xml (hunk 3 above)
Comment (mtekman) [reply tools/rgrnastar/rg_rnaStarSolo.xml]: added

File: tools/rgrnastar/rg_rnaStarSolo.xml (hunk 4 above)
Comment (mtekman) [reply tools/rgrnastar/rg_rnaStarSolo.xml]: added

**Comment by bgruening** [comment]
Awesome work @mtekman @wm75!

**Pull request merged**